download-lambda-logs/download_logs/aws_lambda.py

```diff
@@ -59,14 +60,21 @@ def parse_client_id(self, client_id):
         client_id = client_id.split(".")[-2:]
         return ".".join(client_id)
 
+    def calculate_time_delta(self, timestamp):
+        real_hit_time = datetime.fromtimestamp(int(timestamp))
+        timedelta = datetime.now() - real_hit_time
+        return int(timedelta.total_seconds() * 1000)
+
     def construct_url(self, download_data):
         property_id = 'UA-26179049-7'
         ga_client_id = download_data['ga_client_id'] or 'No client id'
         category = 'Download from External Source'
         filename = download_data['file_downloaded'] or 'No filename present'
         referrer = download_data['referrer'] or 'No referrer'
         user_agent = download_data['user_agent'] or 'No user agent'
-        ip = download_data['ip']
+        ip = download_data['ip'] or 'No IP'
+        timestamp = download_data['timestamp'] or 'No timestamp'
```

Comment:
Supplying a string here will make the script error. It'd be better to supply the current timestamp and default to a zero `qt` parameter.